tests: `pkg.bugtool.TestFindMaps` is flaky

This one's on me 😔 

https://github.com/cilium/tetragon/actions/runs/18681499471/job/53264252392?pr=4216

For reference, this is the test failure log:
```
level=info msg="BTF discovery: default kernel btf file found" btf-file=/sys/kernel/btf/vmlinux
=== RUN   TestFindMaps
=== RUN   TestFindMaps/NoSuchFile
=== RUN   TestFindMaps/BaseSensorMemlock
level=info msg="Exit probe on acct_process"
level=info msg="Set execve_map entries 32768" size=28M
level=info msg="BPF ring buffer size (bytes)" total=128K
level=info msg="BTF file: using metadata file" metadata=/sys/kernel/btf/vmlinux
level=info msg="Loading sensor" name=__base__
level=info msg="Loading kernel version 6.1.155"
level=info msg="Loaded sensor successfully" sensor=__base__
level=info msg="Read ProcFS /proc appended 71/127 entries"
    maps_test.go:48: 
        	Error Trace:	/home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/pkg/bugtool/maps_test.go:48
        	Error:      	Should not be zero, but was 0
        	Test:       	TestFindMaps/BaseSensorMemlock
level=info msg="Unloading sensor __base__"
level=info msg="Sensor unloaded" sensor=__base__ maps-error=[]
    base.go:198: cleanup: unloading base sensor
level=info msg="Unloading sensor __base__"
--- FAIL: TestFindMaps (0.62s)
    --- PASS: TestFindMaps/NoSuchFile (0.00s)
    --- FAIL: TestFindMaps/BaseSensorMemlock (0.62s)
FAIL
```

That means the failure is here:
```go
mapsUsedByProgs, err := FindMapsUsedByPinnedProgs(path)
require.NoError(t, err)
if assert.NotEmpty(t, mapsUsedByProgs) {
	assert.NotZero(t, mapsUsedByProgs[0].Memlock)
}
```

--- a/pkg/bugtool/maps_test.go
+++ b/pkg/bugtool/maps_test.go
@@ -12,6 +12,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"github.com/cilium/tetragon/pkg/bpf"
 	tus "github.com/cilium/tetragon/pkg/testutils/sensors"
 
 	// needed to register the probe type execve for the base sensor
@@ -32,6 +33,14 @@ func TestFindMaps(t *testing.T) {
 		require.Error(t, err)
 	})
 
+	sumMemlock := func(mapInfos []bpf.ExtendedMapInfo) int {
+		sumMemlock := 0
+		for _, mapInfo := range mapInfos {
+			sumMemlock += mapInfo.Memlock
+		}
+		return sumMemlock
+	}
+
 	t.Run("BaseSensorMemlock", func(t *testing.T) {
 		tus.LoadInitialSensor(t)
 
@@ -39,19 +48,19 @@ func TestFindMaps(t *testing.T) {
 		pinnedMaps, err := FindPinnedMaps(path)
 		require.NoError(t, err)
 		if assert.NotEmpty(t, pinnedMaps) {
-			assert.NotZero(t, pinnedMaps[0].Memlock)
+			assert.NotZero(t, sumMemlock(pinnedMaps))
 		}
 
 		mapsUsedByProgs, err := FindMapsUsedByPinnedProgs(path)
 		require.NoError(t, err)
 		if assert.NotEmpty(t, mapsUsedByProgs) {
-			assert.NotZero(t, mapsUsedByProgs[0].Memlock)
+			assert.NotZero(t, sumMemlock(mapsUsedByProgs))
 		}
 
 		allMaps, err := FindAllMaps()
 		require.NoError(t, err)
 		if assert.NotEmpty(t, allMaps) {
-			assert.NotZero(t, allMaps[0].Memlock)
+			assert.NotZero(t, sumMemlock(allMaps))
 		}
 	})
 

